lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -1950,27 +1953,33 @@ def plot_trisurf(self, *args, color=None, norm=None, vmin=None, vmax=None,
         yt = tri.y[triangles]
         zt = z[triangles]
         verts = np.stack((xt, yt, zt), axis=-1)
+        if facecolors is not None:
+            facecolors = facecolors[triangles].mean(axis=1)
 
         polyc = art3d.Poly3DCollection(verts, *args, **kwargs)
 
         if cmap:
             # average over the three points of each triangle
-            avg_z = verts[:, :, 2].mean(axis=1)
-            polyc.set_array(avg_z)
+            if facecolors is not None:
+                polyc.set_array(facecolors)
+            else:
+                avg_z = verts[:, :, 2].mean(axis=1)
+                polyc.set_array(avg_z)
             if vmin is not None or vmax is not None:
                 polyc.set_clim(vmin, vmax)
```

Comment:
This doesn't make sense - why would you set both a colormap and face colors?

Reply:
I was aiming at behaviour equivalent to `plt.scatter`: either have 1 float per face to be mapped using `cmap`, or a full color specification per face, in which case the cmap argument indeed makes no sense. Looking back the code is however not checking this. I will revisit tonight and try to fix conflicts.